assets/performanceprofile/scripts/set-rps-mask.sh

```diff
@@ -1,24 +1,20 @@
 #!/usr/bin/env bash
 
 function set_queue_rps_mask() {
-queue_path=${path%rx*}
-
-queue_num=${path#*queues/}
-
-# replace '/' with '-'
-queue_num="${queue_num/\//-}"
-
+# replace x2d with hyphen (-) which is an escaped character
```

Comment:
does `systemd-escape` support reverse functionality (un-escape) and if so, do we want to use it?

Reply:
It supports reverse.
The original systemd-escape output is:
`\x2d` but the backslash gets removed when we pass the argument to the script. Due to that fact it failed to revert this part.
Besides that, when we call the script we invoked it under the systemd-unit and passing the path argument with `%I` which basically perform the unesacping for us and which is cheaper than fork another process for the systemd-escape call here

Reply:
it surely is, but it's also asymmetric. We do escaping using systemd-escape, we do unescaping ourselves. We save resources, we need to ensure the consistency ourselves.
Now: this can very much be a path we want to choose, but let's make sure it's intentional, agreed, explicit.

Reply:
Yes I agree it's kind of odd and if the de-escaping was perfect I would definitely choose this path.
But I really didn't understand why the backslash get omitted. It's probably related to the way of how the systemd-unit passes the argument

Reply:
possibly true. Considering the nature of this bug I won't block for this, even though I think it would deserve a followup